src/main/java/org/folio/circulation/domain/notice/schedule/DueDateScheduledNoticeHandler.java

```diff
@@ -70,12 +84,46 @@ public DueDateScheduledNoticeHandler(
 
   private CompletableFuture<Result<ScheduledNotice>> handleDueDateNotice(ScheduledNotice notice) {
     return loanRepository.getById(notice.getLoanId())
+      .thenCompose(r -> deleteNoticeIfLoanIsMissingOrIncomplete(r, notice))
       .thenApply(r -> r.map(LoanAndRelatedRecords::new))
       .thenCompose(r -> r.after(loanPolicyRepository::lookupLoanPolicy))
-      .thenCompose(r -> r.combineAfter(configurationRepository::findTimeZoneConfiguration,
-        LoanAndRelatedRecords::withTimeZone))
+      .thenCompose(r -> r.combineAfter(configurationRepository::findTimeZoneConfiguration, LoanAndRelatedRecords::withTimeZone))
       .thenCompose(r -> r.after(records -> sendNotice(records, notice)))
-      .thenCompose(r -> r.after(relatedRecords -> updateNotice(relatedRecords, notice)));
+      .thenCompose(r -> r.after(relatedRecords -> updateNotice(relatedRecords, notice)))
+      .thenApply(r -> r.mapFailure(this::handleFailure));
+  }
+
+  CompletableFuture<Result<Loan>> deleteNoticeIfLoanIsMissingOrIncomplete(
+      Result<Loan> result, ScheduledNotice notice) {
+
+    if (failedToFindRecordOfType(result, LOAN_REC_TYPE)) {
+      return deleteInvalidNoticeAndFail(notice, LOAN_REC_TYPE, notice.getLoanId());
+    }
+
+    if (result.succeeded()) {
+      Loan loan = result.value();
+      if (loan.getItem().isNotFound()) {
+        return deleteInvalidNoticeAndFail(notice, ITEM_REC_TYPE, loan.getItemId());
+      }
+      if (loan.getUser() == null) {
+        return deleteInvalidNoticeAndFail(notice, USER_REC_TYPE, loan.getUserId());
+      }
+    }
+    return completedFuture(result);
+  }
+
+  private CompletableFuture<Result<Loan>> deleteInvalidNoticeAndFail(
+      ScheduledNotice notice, String missingRecordType, String missingRecordId) {
+    log.info(NOTICE_DELETION_MSG, notice.getId(), missingRecordType, missingRecordId);
```

Comment:
It is more clear for me to have explicit logging message instead of extracting it to the constant.

Reply:
good point, done